src/hidra/receiver/asapo_transfer.py

```diff
@@ -121,6 +107,20 @@ def main():
     run_transfer(asapo_transfer, args['reconnect_timeout'])
 
 
+def construct_config(config_file, identifier):
+
+    # Read config file
+    with open(config_file, "r") as f:
+        config = yaml.load(f.read(), Loader=yaml.FullLoader)
+
+    # parse identifier to extruct `beamline` and `detector_id`
+    identifier_info = re.search(".*@(?P<beamline>.*)_(?P<detector_id>.*).service", identifier).groupdict()
```

Comment:
The `@` and `.service` parts are not included. You can just do
```suggestion
    beamline, detector_id = identifier.split("_", maxsplit=1)
```

Reply:
Ok.